apps/website/app/services/referral_service.ts

```diff
@@ -23,24 +25,18 @@ export default class ReferralService {
     return values[0];
   }
 
-  // To be moved to a policy once bouncer is installed
-
   async canUserRefer(user: User) {
-    if (user.isPromoter()) return true;
-
-    return await User.hasPurchasedTicket(user);
+    return await new Bouncer(user).with(UserPolicy).allows("refer");
   }
 
   async canUserBeLinked(user: User) {
-    return !user.isPromoter() && !user.wasReferred() && !(await User.hasPurchasedTicket(user));
+    return await new Bouncer(user).with(UserPolicy).allows("beLinked");
```

Comment:
Maybe a single instance of Bouncer could be used throughout the whole application, instead of one being created for each request? I have been looking at the Bouncer documentation and they do mention DI but it's for injecting values inside policies, not the other way around. Food for thought.

Reply:
A single bouncer instance already exists - it is included in each request. The only reason why I did not use that instance is because we pass a specific user in each function as an argument, and we make these authorization checks against someone other than the logged-in user (referrers, for example). As far as I know, there is no way to replace the user in the existing bouncer instance. If we only performed these authorizations for the currently logged-in user, I could inject the HttpContext in the service and then use `this.ctx.bouncer`.

Reply:
Oh yeah, you're totally right, I missed it the first time I checked Bouncer's documentation. All good on my end then.